### DISCUSSION TITLE
Multiple Users added to user table on publish

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#9816] [Help] [answered]
Hello,

Budibase cloud , Appversion: 2.3.16

Users are added automatically to budibase user table on publishing the app. 

See below , No user was provided access to the app but on publishing the app, multiple user records were added to table.  how can we delete them?  These user should not have access to this app. 

Steps performed:

- Create new app
- Create a external datasource (postgressql) and add query
- Created automation and home page to display data from external datasource query
- publish app
- noticed users in the user table. 
- Expected only the user who published app to be added to users table

![image](https://user-images.githubusercontent.com/42922833/221581576-d5c06686-c47f-4f73-9f2d-bd9303de00ed.png)

![image](https://user-images.githubusercontent.com/42922833/221582078-3bc815f3-cbc7-4dd0-a0e8-5be35676419c.png)

**Comment by melohagan** [comment]
Hey @applicationuser 

All admin and developer users will have access to published apps. 

To remove a user from the Users' table of an app, go to the [User Management](https://docs.budibase.com/docs/user-management) section of the Budibase portal, and click on a user.

This will take you to a detail screen (see the info note for an Admin or Developer user)

![Screenshot 2023-03-01 at 11 44 22](https://user-images.githubusercontent.com/101575380/222130114-40cc90e5-295e-4e74-80e0-324691154ed6.png)

Change the role to **App User**, and then that user will only be able to access apps that they have been assigned to.

**Reply by applicationuser** [reply]
@melohagan 
Thank you for your response.
Not all the users added to user table are Admin/Developer roles user. They were defined with App user role. 
strange thing noticed is all these non admin/developer users assigned to the app were part of User Group.   Group does not have this app assigned.  Not sure how they are getting access to this app .

Thanks

**Comment by melohagan** [comment, marked as answer]
Thanks for highlighting this issue @applicationuser 

I have raised a bug here: https://github.com/Budibase/budibase/issues/9845

It appears that despite appearing in the users table, the users do not have access to the app unless specified in the access list.
